src/silicone/utils.py

```diff
@@ -1,12 +1,223 @@
+import numpy as np
+import pandas as pd
+import scipy.interpolate
+
 """
 Utils contains a number of helpful functions that don't belong elsewhere.
 """
 
 
+def find_matching_scenarios(
```

Comment:
I was hoping that this could be made to work with an arbitrary cruncher (i.e. you would pass the cruncher in as an argument somewhere). The function logic would then be:

1. loop over all scenarios in `classify_scenarios`
1. loop over all modes in `classify_models`
1. get the set of data which matches the relevant scenario and model
1. if there's no data, show warning and go to next option
1. otherwise use the specified cruncher to get the filler function
1. pass the `variable_leaders` data in `to_compare_df` to the filler function just derived to create `infilled` data for this specific subset of scenarios and models
1. take RMSE between `infilled` and the `variable_follower` data in `to_compare_df`

I think my suggestion would be a slight change in logic (you'd change from saying, which scenario-model combo has the most similar points to which scenario-model combo results in a filler which best matches the relationship seen between the leader and follower variables in `to_compare_df` but I think the ranking of which model-scenario combo is best would remain the same).

Reply:
If the above is more mucking around than it's worth then `find_matching_scenarios` can go back into database_crunchers/ssp_specific_relation.py as it is database cruncher specific.

Reply:
Let's discuss this more later, but conceptually I don't know why this would be useful for other models. The other models generally behave more regularly when you add more data, whereas the interpolation behaves more irregularly, hence the desire to filter it down. I could code up what you've asked for but I don't know that anyone would particularly want to use it.